form-flow/src/main/kotlin/com/ritense/formflow/service/FormFlowService.kt

```diff
@@ -36,6 +38,18 @@ class FormFlowService(
         return formFlowDefinitionRepository.getById(formFlowDefinitionId)
     }
 
+    fun open(formFlowInstanceId: FormFlowInstanceId) {
+        val formFlowInstance = formFlowInstanceRepository.getById(formFlowInstanceId)
+        val currentStep = formFlowInstance.getCurrentStep()
+        val stepDefinition = currentStep.definition
+
+        val expressionProcessor = expressionProcessorFactory.create(currentStep)
```

Comment:
The current step holds a reference to the formFlowInstance, which in turn has the history.

Reply:
Fair, Then it makes sense to have a method in FormFlowStepInstance that can retrieve the (relevant) history, rather than exposing the entirety of the instance (and as a result things like definition too).